mule-extensions-api/src/main/java/org/mule/runtime/extension/api/model/AbstractExecutableComponentModel.java

```diff
@@ -82,21 +83,22 @@ protected AbstractExecutableComponentModel(String name,
   /**
    * Creates a new instance
    *
-   * @param name                 the model's name
-   * @param description          the model's description
+   * @param name the model's name
+   * @param description the model's description
    * @param parameterGroupModels a {@link List} with the source's {@link ParameterGroupModel parameter group models}
-   * @param output               an {@link OutputModel} which represents the component's output content
-   * @param outputAttributes     an {@link OutputModel} which represents the component's attributes on the output {@link Message}
-   * @param requiresConnection   whether this component requires connectivity
-   * @param transactional        whether this component supports transactions
-   * @param supportsStreaming    whether this component supports streaming
-   * @param displayModel         a model which contains directive about how this component is displayed in the UI
-   * @param stereotype           the {@link StereotypeModel stereotype} of this component
-   * @param modelProperties      A {@link Set} of custom properties which extend this model
-   * @param nestedComponents     a {@link List} with the components contained by this model
-   * @param notifications        A {@link Set} of {@link NotificationModel} which describes the component's notifications
+   * @param output an {@link OutputModel} which represents the component's output content
+   * @param outputAttributes an {@link OutputModel} which represents the component's attributes on the output {@link Message}
+   * @param requiresConnection whether this component requires connectivity
+   * @param transactional whether this component supports transactions
+   * @param supportsStreaming whether this component supports streaming
+   * @param displayModel a model which contains directive about how this component is displayed in the UI
+   * @param stereotype the {@link StereotypeModel stereotype} of this component
+   * @param modelProperties A {@link Set} of custom properties which extend this model
+   * @param nestedComponents a {@link List} with the components contained by this model
+   * @param notifications A {@link Set} of {@link NotificationModel} which describes the component's notifications
    * @throws IllegalArgumentException if {@code name} is blank
    */
+  @Deprecated
```

Comment:
there's no need to deprecate all these constructors. It's actually good to have smaller constructors with default values.

Reply:
👍